### ISSUE TITLE
C# method GPURender.Resize causes fatal C++ Error when using invalid parameters

### ISSUE THREAD

**Issue by Pikanchion** [FlaxEngine/FlaxEngine#2261]
**Issue description:**
Calling the C# method `GPURender.Resize` with an invalid `width` or `height` causes a C++ Assertion Error (`GPUTexture.cpp` Line 368), crashing the Editor. This not only has the potential to be misinterpreted as an Engine bug (as opposed to if the call were prevented or otherwise reverted, and displayed instead as an error in the Debug/Output Log), but also greatly increases the difficulty in debugging the root cause in cases where the erroneous C# call to `Resize` is made only under some novel circumstance, given the logged stack trace only goes as far as the offending interop call.

**Steps to reproduce:**
1. Create a `GPURender`.
2. Call `GPURender.Resize(0, 0)` or similar.

**Flax version:**
1.7.2

**Comment by Pikanchion** [comment]
Incidentally, is it intentional that the `width`/`height` of a `GPURender` may not be set to `1`?

**Comment by mafiesto4** [comment]
Fixed in 1cb88bdc63ae9447e1c72305f60cf3cf864b9a46.

**Comment by Pikanchion** [comment]
@mafiesto4 This fixes cases where either `width` or `height` less than `1`, however, there are still two issues present:

1. When **both** values are greater than `0` and **at least one** is equal to `1`, there is an `Unhandled exception: EXCEPTION_ACCESS_VIOLATION reading address 0x54`.

```
Stack trace:
    at FlaxEditor.exe!GPUResource::SetName() in FlaxEngine\Source\Engine\Graphics\GPUDevice.cpp:line 239
    at FlaxEditor.exe!ScreenSpaceReflectionsPass::Render() in FlaxEngine\Source\Engine\Renderer\ScreenSpaceReflectionsPass.cpp:line 194
    at FlaxEditor.exe!ReflectionsPass::Render() in FlaxEngine\Source\Engine\Renderer\ReflectionsPass.cpp:line 452
    at FlaxEditor.exe!RenderInner() in FlaxEngine\Source\Engine\Renderer\Renderer.cpp:line 517
    at FlaxEditor.exe!Renderer::Render() in FlaxEngine\Source\Engine\Renderer\Renderer.cpp:line 213
    at FlaxEditor.exe!SceneRenderTask::OnRender() in FlaxEngine\Source\Engine\Graphics\RenderTask.cpp:line 391
    at FlaxEditor.exe!RenderTask::OnDraw() in FlaxEngine\Source\Engine\Graphics\RenderTask.cpp:line 84
    at FlaxEditor.exe!RenderTask::DrawAll() in FlaxEngine\Source\Engine\Graphics\RenderTask.cpp:line 46
    at FlaxEditor.exe!GPUDevice::Draw() in FlaxEngine\Source\Engine\Graphics\GPUDevice.cpp:line 632
    at FlaxEditor.exe!Engine::OnDraw() in FlaxEngine\Source\Engine\Engine\Engine.cpp:line 359
    at FlaxEditor.exe!Engine::Main() in FlaxEngine\Source\Engine\Engine\Engine.cpp:line 214
    at FlaxEditor.exe!wWinMain() in FlaxEngine\Source\Engine\Main\Windows\main.cpp:line 55
    at FlaxEditor.exe!__scrt_common_main_seh() in D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:line 288
    at KERNEL32.DLL 0x7ffa590e7344
    at ntdll.dll 0x7ffa594026b1
```
2. When **both** values are greater than `1` and **both** are less than `8`, there is an `[Error] Critical error! Reason: Assertion failed!` in `FlaxEngine\Source\Engine\Core\Collections\Array.h`, Line: 315: `Expression: index >= 0 && index < _count`.
```
Stack trace:
    at FlaxEditor.exe!Log::Logger::Write() in FlaxEngine\Source\Engine\Core\Log.cpp:line 260
    at FlaxEditor.exe!PlatformBase::Assert() in FlaxEngine\Source\Engine\Platform\Base\PlatformBase.cpp:line 451
    at FlaxEditor.exe!GPUTextureDX11::View() in FlaxEngine\Source\Engine\GraphicsDevice\DirectX\DX11\GPUTextureDX11.h:line 242
    at FlaxEditor.exe!PostProcessingPass::Render() in FlaxEngine\Source\Engine\Renderer\PostProcessingPass.cpp:line 320
    at FlaxEditor.exe!RenderInner() in FlaxEngine\Source\Engine\Renderer\Renderer.cpp:line 618
    at FlaxEditor.exe!Renderer::Render() in FlaxEngine\Source\Engine\Renderer\Renderer.cpp:line 213
    at FlaxEditor.exe!SceneRenderTask::OnRender() in FlaxEngine\Source\Engine\Graphics\RenderTask.cpp:line 391
    at FlaxEditor.exe!RenderTask::OnDraw() in FlaxEngine\Source\Engine\Graphics\RenderTask.cpp:line 84
    at FlaxEditor.exe!RenderTask::DrawAll() in FlaxEngine\Source\Engine\Graphics\RenderTask.cpp:line 46
    at FlaxEditor.exe!GPUDevice::Draw() in FlaxEngine\Source\Engine\Graphics\GPUDevice.cpp:line 632
    at FlaxEditor.exe!Engine::OnDraw() in FlaxEngine\Source\Engine\Engine\Engine.cpp:line 359
    at FlaxEditor.exe!Engine::Main() in FlaxEngine\Source\Engine\Engine\Engine.cpp:line 214
    at FlaxEditor.exe!wWinMain() in FlaxEngine\Source\Engine\Main\Windows\main.cpp:line 55
    at FlaxEditor.exe!__scrt_common_main_seh() in D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:line 288
    at KERNEL32.DLL 0x7ffa590e7344
    at ntdll.dll 0x7ffa594026b1
```

**Comment by mafiesto4** [comment]
Fixed in 149a6a29f883e83a3763eb92328cc66dc254f534.
